bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/BroadlinkHandlerFactory.java

```diff
@@ -0,0 +1,83 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.broadlinkthermostat.internal.handler.FloureonThermostatHandler;
+import org.openhab.binding.broadlinkthermostat.internal.handler.RMUniversalRemoteHandler;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BroadlinkHandlerFactory} is responsible for creating things and thing handlers.
+ *
+ * @author Florian Mueller - Initial contribution
+ */
+@Component(configurationPid = "binding.broadlinkthermostat", service = ThingHandlerFactory.class)
+@NonNullByDefault
+public class BroadlinkHandlerFactory extends BaseThingHandlerFactory {
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(
+            BroadlinkBindingConstants.FLOUREON_THERMOSTAT_THING_TYPE,
+            BroadlinkBindingConstants.RM_UNIVERSAL_REMOTE_THING_TYPE,
+            BroadlinkBindingConstants.HYSEN_THERMOSTAT_THING_TYPE,
+            BroadlinkBindingConstants.UNKNOWN_BROADLINK_THING_TYPE,
+            BroadlinkBindingConstants.UNKNOWN_BROADLINK_THERMOSTAT_THING_TYPE);
+    private static final String BROADLINK_FOLDER = Path.of(OpenHAB.getUserDataFolder(), "broadlink").toString();
+    public static final String INFRARED_FOLDER = Path.of(BROADLINK_FOLDER, "infrared_commands").toString();
+    static {
+        Logger logger = LoggerFactory.getLogger(BroadlinkHandlerFactory.class);
+        File directory = new File(BROADLINK_FOLDER);
+        if (!directory.exists()) {
+            if (directory.mkdir()) {
+                logger.info("broadlink dir created {}", BROADLINK_FOLDER);
+            }
+        }
+        File childDirectory = new File(INFRARED_FOLDER);
+        if (!childDirectory.exists()) {
+            if (childDirectory.mkdir()) {
+                logger.info("infrared_commands dir created {}", INFRARED_FOLDER);
+            }
+        }
+    }
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES.contains(thingTypeUID);
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (BroadlinkBindingConstants.FLOUREON_THERMOSTAT_THING_TYPE.equals(thingTypeUID)
+                || BroadlinkBindingConstants.HYSEN_THERMOSTAT_THING_TYPE.equals(thingTypeUID)) {
+            return new FloureonThermostatHandler(thing);
+        }
+        if (BroadlinkBindingConstants.RM_UNIVERSAL_REMOTE_THING_TYPE.equals(thingTypeUID)) {
+            return new RMUniversalRemoteHandler(thing);
+        }
+        return null;
```

Comment:
What for the 2 unknown thing types ?
As they are supported, you must create an handler for them. If not, the things will remain uninitialized. Maybe it is what you expect ?

Reply:
But I see no real interest to create things if they are not handled by the binding.

Reply:
Yes, that is from the original implementation, it was also strange for me.
I think it just to show you there are other devices compatibles with the protocol but not supported, I think that by returning null thing remains uninitialized.
Not sure what to do, I can completely remove the unknown thing types.

Reply:
Yes, I think it would be better to completely remove the unknown thing types.